config/webpack.config.js

```diff
@@ -33,7 +44,71 @@ const globalEntryFiles = {
   ],
 };
 
-const configGenerator = (buildOptions, apps) => {
+/**
+ * Get a list of all the entry points.
+ *
+ * If the `entry` CLI argument is passed, only the specified
+ * application entries are built.
+ */
+function getEntryPoints(entry) {
+  const manifests = getAppManifests();
+  let manifestsToBuild = manifests;
+  if (entry) {
+    const entryNames = entry.split(',').map(name => name.trim());
+    manifestsToBuild = manifests.filter(manifest =>
+      entryNames.includes(manifest.entryName),
+    );
+  }
+
+  return getWebpackEntryPoints(manifestsToBuild);
+}
+
+/**
+ * The dev server requires settings.js for building the
+ * redirects. This loads the settings for use in the watch commands.
+ *
+ * This does NOT recreate the full settings like
+ * create-build-settings.js does. This is currently only meant to fill
+ * out the URL rewrites for the webpack-dev-server.
+ *
+ * @return {Object} settings
+ */
+function getSettings() {
+  const settings = {};
+  const manifests = getAppManifests();
+  settings.applications = manifests
+    .map(
+      m =>
+        // Some manifests don't have a rootUrl
+        m.rootUrl && {
+          contentProps: [
+            {
+              path: path.join('.', m.rootUrl),
+            },
+          ],
+        },
+    )
+    // Filter out empty entries
+    .filter(a => !!a);
```

Comment:
Could we filter first and then map? I think this might remove the need for the ternary.